### ISSUE TITLE
installation of repository

### ISSUE THREAD

**Issue by jorgerosas87** [kprussing/cloudgen#1]
the following error:
"CMake Error at CMakeLists.txt:56 (add_executable):
Target "cloudgen" links to target "netCDF::netcdf" but the target was not
found. Perhaps a find_package() call is missing for an IMPORTED target, or
an ALIAS target is missing?"

By the way I have installed the netcdf4 package from https://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf-c-4.8.0.tar.gz

**Comment by jorgerosas87** [comment]
Do you have any docker or Virtual PC with the software already installed??

**Comment by jorgerosas87** [comment]
I  have still the same problem: "Segmentation Fault", I made the installation in another PC (Ubuntu 18.04 ) .

**Comment by kprussing** [comment]
First some administrative details, I just pushed a change and made `trunk` the default branch so you may need to `git pull` if you wind up with a fix. The change shouldn't affect your issue because you already had the upstream patch.

Second, so you have been able to successfully build the software (as in it does find netCDF correctly)? What command are you running and with what configuration file? My tests aren't showing an issue with a segmentation fault.

**Comment by jorgerosas87** [comment]
I installed by mkdir build     cmake ../    sudo make sudo make install
Previoulsy ,  I installed fftw3 and netdf library ...
I made the pull and entering into trunk branch and I did the same procedure,  with still the same error of "Segmentation Fault (core dumped)". Are you any docker made of this software you can please give me ? or any virtual PC??

**Comment by kprussing** [comment]
No, I don't have a docker or anything like that because I don't use docker. I install netCDF4, FFTW and the like from the appropriate package managers or from source on the machines on which I don't have admin privileges. My suspicion is a bad dependency on your install.

1.  *How* did you install `netCDF4` and `FFTW`? Was it from source, a package manager, etc.?
    1.  If from source, what version?
    2. If from a package manager, what platform, package manager, and version?
2.  Are the correct libraries linked to the executable? (i.e. what does `ldd <path/to/>cloudgen` say? Are the library linkages correct?)
3.  What particular input file are you using? Is it valid? (in fact, this could be the ultimate source of the error).

A key point: This is research grade software. It is not necessarily production grade, and, therefore, it assumes a bit of effort on the end users part (thus the CMake build I put together to make my life a bit easier) which includes potentially installing dependencies by hand.

**Comment by jorgerosas87** [comment]
1) We test with two users  in one machine and a Virtual Machine (using FTW is version 3.3.9):  
1-we used netcdf 4.1.3 then compiled netcdf with gcc / gcc after, we used cmake to compile cloudgen using gcc/gfortran,  
2-In the second user in the same machine we use netcdf 4.1.3  with ifort /icc and compile cloudgen with ifort/icc  
3-In a virtual Pc we used netcdf library version 4.8.0 , following the cmake procedures .

* we don't use any  package manager, we download fftw and nectdef from the oficial web sites...

2) The LD_LIBRARY_PATH and  PATH  are correct, but we realize that cloudgen has its own netcdf library.
We install cloudgen for one user in /usr/local/bin/cloud-intel/ and in this moment the netcdf  PATH is /usr/local/bin/cloud-intel/bin/ncgen and version is "netcdf library version 4.7.3-development." 
Thus , there are different netcdfs  -- netcdf 4.1.3  used to create cloudgen with cmake and other netcdf  generated by cmake and cloudgen code -- version 4.7.3

3) The inputs" *.dat"  used are included in the "samples" folder:
For example: 
cloudgen stratocumulus.dat : to run for the configuration of stratocumulus and so on other cases such as cirrus ..

Thanks in advance for your help,  and we understand  that hand  installation may be necessary  , 
Finally, we have some issues:
which gcc/gfortran  or icc/ifort do you use  ?
Which netcdf version is better for use cmake in this case ?

**Comment by kprussing** [comment]
Great. I use NetCDF 4.8.0 (which has a patch to the CMakeLists.txt that was merged back in November 2019) and FFTW 2.15 (with the FindFFTW.cmake after October 2019 with another merged patch). For me, both NetCDF and FFTW are from macports (on MacOS).

In theory, the build should respect any installed version of FFTW and NetCDF if it can find them.

If I use the default compiler on MacOS (`cc`), I have no issue. However, if I set `CC` to `clang` or `gcc` (from macports), I do get the segmentation fault. I don't have a version of the Intel compilers handy so I can't speak to that. This is new to me, so I'll see if I can figure out what is going on.

*edit*: Or not. It appears to happen intermittently for me with all compilers. And it appears to vanish if I run it in the debugger...
